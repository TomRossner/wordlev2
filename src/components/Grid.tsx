import React, { useEffect } from 'react';
import Tile from './Tile.tsx';
import { IGrid, ITile } from '../interfaces';

interface GridProps {
    grid: IGrid;
    resetGrid: () => void;
    handleRemoveShake: () => void;
}

const Grid: React.FC<GridProps> = ({grid, resetGrid, handleRemoveShake}) => {
    

    // Create grid
    useEffect(() => {
        resetGrid();
    }, []);

  return (
    <section id='grid' className='tiles-container'>
        {grid.grid.map((row: ITile[], rowIndex: number) => {
            return (
                <div key={rowIndex} className='row'>
                    {row.map((tile: ITile, tileIndex: number) => {
                        return (
                            <Tile key={tileIndex} tile={tile} handleRemoveShake={handleRemoveShake}/>
                        )
                    })}
                </div>  
            )
        })}
    </section>
  )
}

export default Grid;