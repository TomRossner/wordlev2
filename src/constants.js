export const WORD_LENGTH = 5;

export const MAX_GUESSES = 6;

export const COLORS = {
    GREEN: 'green',
    YELLOW: 'yellow',
    GRAY: 'gray',
}