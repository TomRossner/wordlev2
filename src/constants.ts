import { ITile, IColor } from "./interfaces";

export const WORD_LENGTH: number = 5;

export const MAX_GUESSES: number = 6;

export const LETTER_REGEX: RegExp = /^[a-zA-Z]$/;

export const TILE: ITile = {
    letter: '',
    colorClass: ''
}

export const COLORS: IColor = {
    GREEN: 'green',
    YELLOW: 'yellow',
    GRAY: 'gray',
}

export const keyboardLetters: string[] = [
    'q',
    'w',
    'e',
    'r',
    't',
    'y',
    'u',
    'i',
    'o',
    'p',
    '',
    'a',
    's',
    'd',
    'f',
    'g',
    'h',
    'j',
    'k',
    'l',
    '',
    'del',
    'z',
    'x',
    'c',
    'v',
    'b',
    'n',
    'm',
    'enter'
]

// Audio file paths
export const winAudioFilePath: string = process.env.PUBLIC_URL + '/audio/winAudio.mp3';
export const gameOverAudioFilePath: string = process.env.PUBLIC_URL + '/audio/GameOverSound.mp3';
export const emptyAudioFilePath: string = process.env.PUBLIC_URL + '/audio/emptyAudio.mp3';
export const popAudioFilePath: string = process.env.PUBLIC_URL + '/audio/PopSound.mp3';
export const invalidAudioFilePath: string = process.env.PUBLIC_URL + '/audio/invalidAudio.mp3';

// Audio
export const winAudio: HTMLAudioElement = new Audio(winAudioFilePath);
export const gameOverAudio: HTMLAudioElement = new Audio(gameOverAudioFilePath);
export const emptyAudio: HTMLAudioElement = new Audio(emptyAudioFilePath);
export const popAudio: HTMLAudioElement = new Audio(popAudioFilePath);
export const invalidAudio: HTMLAudioElement = new Audio(invalidAudioFilePath);