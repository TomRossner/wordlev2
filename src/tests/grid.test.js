import renderer from 'react-test-renderer';
import Grid from '../components/Grid';
import { MAX_GUESSES, WORD_LENGTH } from '../constants';

const createGrid = (rows, cols) => {
    const newGrid = [];

    for (let i = 0; i < rows; i++) {
      const row = [];

      for (let j = 0; j < cols; j++) {
        row.push(tileMock);
      }

      newGrid.push(row);
    }

    return newGrid;
};

const tileMock = {
    letter: "",
    colorClass: "",
}

const gridMock = {
    grid: createGrid(MAX_GUESSES, WORD_LENGTH)
}

it('Renders correctly', () => {
    const grid = renderer.create(<Grid grid={gridMock}/>).toJSON();

    expect(grid).toMatchSnapshot();
});

