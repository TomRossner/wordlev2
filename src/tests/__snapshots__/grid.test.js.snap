// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<section
  className="tiles-container"
  id="grid"
>
  <div
    className="row"
  >
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
  </div>
  <div
    className="row"
  >
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
  </div>
  <div
    className="row"
  >
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
  </div>
  <div
    className="row"
  >
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
  </div>
  <div
    className="row"
  >
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
  </div>
  <div
    className="row"
  >
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
    <div
      className="tile "
    />
  </div>
</section>
`;
